ramips: add support for EnGenius ESR600

nbriggs opened the pull request:
The EnGenius ESR600 is a dual band wireless router with a 4-port gigabit
Ethernet switch, a gigabit Ethernet WAN port and a USB port.

Specification:

- Bootloader:	U-Boot
- SoC:		MediaTek MT7620A (600 MHz)
- Flash:	16MB, Macronix MX25L12845E
- RAM:		64MB, Nanya NT5TU32M16DG-AC
- Serial:	115200 baud, no header, 3.3V
  		J2: Vcc (arrow), Gnd, Tx, Rx
- USB:		USB 2, 5V
- Ethernet:	5 x 1 Gb/s 4 LAN 1 WAN, Atheros AR8327
- WiFi0:	5 GHz 802.11 b/g/n Ralink RT5592N
  		300 Mb/s, 2T2R
- WiFi1:	2.4 GHz 802.11 b/g/n integrated
  		300 Mb/s, 2T2R
- Antennas:	2 per radio, internal
- LEDs:		1 programmable power (amber)
  		2 programable radio (blue)
		1 programable WPS-5G (blue)
		1 non-programable WAN activity (blue)
		1 unconfigured WPS-2.4G (amber)
- Buttons:	GPIO: Reset, WPS

Installation:

Use the OEM web interface to install the ...-factory.dlf image.
Use the OpenWRT ...-sysupgrade.bin image for future upgrades.

The J2 serial port can be accessed either by soldering in a header,
standard 0.1" spacing, or by using pogo-pins against the back side.

As configured by the OEM, the U-Boot boot delay is short, however quickly
typing "1" leads to the U-Boot "System load Linux to SDRAM via TFTP"
prompt.  The TFTP client is configured by default with its own address as
192.168.99.9,  server as 192.168.99.8, and file name as "uImageESR600"
It will load an OpenWRT initramfs kernel with this method.

Known issues:

1) Only the ports externally labeled WAN, LAN3 and LAN4 are operational.
LAN1 and LAN2 do not appear to power up. This issue is also present
in the Lava LR25G001 which uses the same/similar Atheros switch.

2) The amber WPS-2.4G LED (in the same light guide as the
blue WPS-5G LED) is not configured in the DTS.

3) The blue WAN activity LED is not present in the DTS.
Configuring it as GPIO driven causes the AR8327 switch to
fail to initialize.

Signed-off-by: Nick Briggs <nicholas.h.briggs@gmail.com>

nbriggs commented:
Is there anything else I need to do in order to move this along?
I have verified that the changes still work when reapplied against the current head of the git.openwrt.org master branch.   Thanks.

adrianschmutzler commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -275,6 +275,19 @@ define Device/e1700
+  IMAGES += factory.dlf
+  IMAGE/factory.dlf := $$(sysupgrade_bin) | check-size $$$$(IMAGE_SIZE) | \
+	senao-header -r 0x101 -p 0x57 -t 2
+  SUPPORTED_DEVICES += esr600
```
SUPPORTED_DEVICES is not required here.

nbriggs replied:
Removed SUPPORTED_DEVICES

adrianschmutzler commented on target/linux/ramips/base-files/lib/ramips.sh:
```diff
@@ -172,6 +172,9 @@ ramips_board_detect() {
 		;;
+	*"ESR600")
+		name="esr600"
+		;;
```
This whole case can be removed.

nbriggs replied:
Removed this case, which means that the file no longer has any ESR600 specific changes

nbriggs commented:
Force-pushed the revised commit.  Would it make things easier to deal with if I canceled this pull-request and recreated it against the current openwrt:master state?  Sorry if I made a mess of this!

adrianschmutzler commented on target/linux/ramips/base-files/etc/board.d/01_leds:
```diff
@@ -152,6 +152,10 @@ dir-860l-b1)
 edimax,br-6478ac-v2)
 	set_wifi_led "$boardname:blue:wlan"
 	;;
+esr600)
```
Needs to be "engenius,esr600". The others, too.

nbriggs replied:
Oops. Fixing it there and in the 02_network cases.

adrianschmutzler commented on target/linux/ramips/dts/ESR600.dts:
```diff
@@ -0,0 +1,211 @@
+			};
+
+			iNIC_rf: partition@50000 {
+				label = "iNIC_rf";
```
Consider using lower case here and for the node name.

nbriggs replied:
I would have, but the "iNIC_rf" mixed case partition name was what the OEM was using, and the mixed case usage was present in the WN-AC1167GR, WN-AC733GR3, WN-GX300GR, and WN-AX1167GR DTSs, while it appears nowhere else in lowercase.

adrianschmutzler commented:
> Force-pushed the revised commit. Would it make things easier to deal with if I canceled this pull-request and recreated it against the current openwrt:master state? Sorry if I made a mess of this

Please do not recreate a new PR. This PR is already set up against (maybe older) master. To build against newer master, just pull in the latest changes (git fetch <upstreamremote> master) into your local repo and then rebase your branch against that.
The PR can then just be updated by a forceful push (git push -f <yourgithubremote> <yourbranchname>.

I'm just not sure how to deal with the merge commits in this case, 'cause I newer use them ...

nbriggs commented:
> Please do not recreate a new PR.
OK.  I'm hoping that this PR/my branch is currently in a usable state, and that perhaps there are no more changes.

adrianschmutzler commented:
There has been a reorganization in ramips target today. Please:
- update your board.d subfiles since they have been re-sorted
- update your Makefile (mt****/rt****.mk) changes so sorting of devices is correct
- rename your DTS/DTSI files following the scheme soc_vendor_device.dts(i), e.g. mt7620a_tplink_archer-c50-v1.dts (and update includes there if necessary)
- remove `DTS := ...` from Makefile
- add `MTK_SOC` to Makefile (needs to match "soc" in DTS file name), e.g. `MTK_SOC := mt7620a`

Thank you!

You might take the chance to squash all your changes into a single commit.

nbriggs commented:
I understand the actual changes in the files that I need to make, but I'm unsure what the right way to get my branch rebased to the correct starting point is.  Hints?  It would be great if https://openwrt.org/docs/guide-developer/working-with-github-pr had notes on what to do when the repo/branch your PR is based on gets updated...

nbriggs commented:
Brought up to date with the revised ramips target, per request.
Squashed to single commit.

adrianschmutzler commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -362,6 +362,20 @@ define Device/elecom_wrh-300cr
+  IMAGES += factory.dlf
+  IMAGE/factory.dlf := $$(sysupgrade_bin) | check-size $$$$(IMAGE_SIZE) | \
+       senao-header -r 0x101 -p 0x57 -t 2
+  SUPPORTED_DEVICES += esr600
```
You should not need this when you are adding a new device.

nbriggs replied:
Agreed.  Pushed commit to remove manipulation of SUPPORTED_DEVICES.

ynezz commented:
>  Squashed to single commit.

Nope, there are still two.

nbriggs commented:
Yeah, there are now two -- I'll squash again.  I was waiting to see if there were more changes (other than squashing) required.

ynezz commented on target/linux/ramips/dts/mt7620a_engenius_esr600.dts:
```diff
@@ -0,0 +1,211 @@
+	};
+
+	leds {
+		// FIXME: locate WPS 5g blue LED
```
You can remove the comments in the DTS and move them to the commit message/description instead.

nbriggs replied:
Yes.

ynezz commented on target/linux/ramips/dts/mt7620a_engenius_esr600.dts:
```diff
@@ -0,0 +1,211 @@
+		led_power: power {
+			label = "esr600:amber:power";
+			gpios = <&gpio0 9 GPIO_ACTIVE_LOW>;
+		};
```
Please add new line after every LED node.

nbriggs replied:
OK.

ynezz commented on target/linux/ramips/dts/mt7620a_engenius_esr600.dts:
```diff
@@ -0,0 +1,211 @@
+
+&gpio1 {
+	status = "okay";
+};
```
Separate the nodes by newline here as well.

nbriggs replied:
OK.  Is there a suggested format for DTS files document somewhere that I missed?

ynezz commented on target/linux/ramips/dts/mt7620a_engenius_esr600.dts:
```diff
@@ -0,0 +1,211 @@
+};
+&gpio2 {
+	status = "okay";
+};
```
Separate the nodes by newline here as well.

nbriggs replied:
OK.

ynezz commented on target/linux/ramips/dts/mt7620a_engenius_esr600.dts:
```diff
@@ -0,0 +1,211 @@
+	mtd-mac-address = <&factory 0x4>;
+	mtd-mac-address-increment = <2>;
+
+        port@5 {
```
Some whitespace issue here?

ynezz replied:
Where is the reg property?

nbriggs replied:
Doesn't it just merge the new properties into the existing port@5 in the ethernet definition?  Or does it do a complete replacement of the existing port@5 by this (new) port@5 -- in which case I do indeed need everything that's already in the ethernet/port@5 from mt7620a.dtsi, and likewise for the mdio-bus definition, for mdio-bus/#address-cells and #size-cells.

nbriggs replied:
I've verified that the dtc compiler does what I expected and merges the nodes under the ethernet node in the mt7620a.dtsi, resulting in subnodes that look as below, do you still require the change?  (I am certainly fixing the whitespace issues though)

		port@4 {
			compatible = "mediatek,mt7620a-gsw-port", "mediatek,eth-port";
			reg = <0x4>;
			status = "disabled";
		};

		port@5 {
			compatible = "mediatek,mt7620a-gsw-port", "mediatek,eth-port";
			reg = <0x5>;
			status = "okay";
			phy-mode = "rgmii";
			mediatek,fixed-link = <0x3e8 0x1 0x1 0x1>;
		};

		mdio-bus {
			#address-cells = <0x1>;
			#size-cells = <0x0>;
			status = "okay";
			mediatek,mdio-mode;

			ethernet-phy@0 {
				reg = <0x0>;
				phy-mode = "rgmii";
				qca,ar8327-initvals = <0x10 0x40000000 0x4 0x7600000 0x7c 0x7e 0xc 0x5600000 0x94 0x7e>;
			};
		};

ynezz commented on target/linux/ramips/dts/mt7620a_engenius_esr600.dts:
```diff
@@ -0,0 +1,211 @@
+&pcie0 {
+	wifi@0,0 {
+		compatible = "pci1814,5592";
+		reg = <0x10000 0 0 0 0>;
```
0x10000 ?

nbriggs replied:
I agree, the correct value is probably 0x0.

adrianschmutzler commented:
The MAC address assignment of this device puzzles me:
&ethernet: <&factory 0x4> +2
&pcie0: <&factory 0x4>
&wmac: <&iNIC_rf 0x4>
lan_mac=$(mtd_get_mac_ascii u-boot-env ethaddr)
wan_mac=$(mtd_get_mac_ascii u-boot-env wanaddr)

Where is <&factory 0x4> +1?

Why do you assign addresses with different mechanisms for &ethernet and lan_mac?

nbriggs commented:
@adrianschmutzler  -- because I'm learning about OpenWRT and I couldn't find useful documentation/guidance in that area, so I'll take any advice you want to give...

For a device with label MAC address 88:DC:96:03:B7:7A, with 2.4G SSID ...03C254, 5G SSID ...03C255, the interfaces as they come up under the OEM firmware look like:
```
br0       Link encap:Ethernet  HWaddr 88:DC:96:03:C2:54
br2       Link encap:Ethernet  HWaddr 7A:33:90:3F:28:AF
eth2      Link encap:Ethernet  HWaddr 88:DC:96:03:C2:54  
eth2.1    Link encap:Ethernet  HWaddr 88:DC:96:03:C2:54  
eth2.2    Link encap:Ethernet  HWaddr 88:DC:96:03:B7:7A  ** label
ra0       Link encap:Ethernet  HWaddr 88:DC:96:03:C2:55  
rai0      Link encap:Ethernet  HWaddr 88:DC:96:03:C2:54  
rai1      Link encap:Ethernet  HWaddr 8A:DC:96:00:C2:54  
rai2      Link encap:Ethernet  HWaddr 8A:DC:96:01:C2:54  
rai3      Link encap:Ethernet  HWaddr 8A:DC:96:02:C2:54  
wds0      Link encap:Ethernet  HWaddr 88:DC:96:03:C2:55  
wds1      Link encap:Ethernet  HWaddr 88:DC:96:03:C2:55  
wds2      Link encap:Ethernet  HWaddr 88:DC:96:03:C2:55  
wds3      Link encap:Ethernet  HWaddr 88:DC:96:03:C2:55  
wds4      Link encap:Ethernet  HWaddr 88:DC:96:03:C2:54  
wds5      Link encap:Ethernet  HWaddr 88:DC:96:03:C2:54  
wds6      Link encap:Ethernet  HWaddr 88:DC:96:03:C2:54  
wds7      Link encap:Ethernet  HWaddr 88:DC:96:03:C2:54  
```
The u-boot-env (Config) partition has these address related properties:
```
ethaddr="88:DC:96:03:C2:54"
wanaddr=88:DC:96:03:B7:7A
wlanaddr=00:AA:BB:CC:DD:12
iNICaddr=00:AA:BB:CC:DD:22
```
So it was not clear to me where the "truth" for the addresses lives -- in the u-boot-env partition? clearly not for the two radio addresses as expressed in wlanaddr and iNICaddr, so in the radio firmware partitions (at offset 4) for the radios -- though the ethaddr property corresponds to the rai0 (and ra0 is derived from rai0+1).

adrianschmutzler commented:
Okay, so you have three "real" addresses:
88:DC:96:03:C2:54 (ethaddr)
88:DC:96:03:C2:55 (calculated?)
88:DC:96:03:B7:7A (wanaddr)

The addresses starting with 8A and 7A are local (local bit set).

So for me, I'd say the truth lies in the u-boot-env partition.

However, we've not learned anything about the other partitions so far (<&factory 0x4>, <&iNIC_rf 0x4>). So, it would be interesting if you did:
```
. /lib/functions/system.sh
get_mac_binary factory 4
get_mac_binary "iNIC_rf" 4
```
(This pastes the data without newline, so you might want to produce a newline between both get_mac_binary calls.)

If this returns MAC addresses, please post them. If this returns anything else, then your mtd-mac-address entries are probably wrong.

nbriggs commented:
```
# get_mac_binary /dev/mtd2ro 4
88:dc:96:03:c2:55
# get_mac_binary /dev/mtd3ro 4
88:dc:96:03:c2:54
```
I had derived the offset 4 directly from the factory and iNIC_rf partitions as I dumped them before doing anything else:
```
mtd2-factory.hex:00000000  92 55 07 01 88 dc 96 03  c2 55 92 55 14 18 01 80  |.U.......U.U....|
mtd3-iNIC_rf.hex:00000000  20 76 04 01 88 dc 96 03  c2 54 ff ff ff ff ff ff  | v.......T......|
```

adrianschmutzler commented:
Okay, I see.
I did not expect the same MAC addresses to be stored twice at different locations.
This means that WiFi in DTS and wan_mac are set correctly.

However, I do not understand the
```
	mtd-mac-address = <&factory 0x4>;
	mtd-mac-address-increment = <2>;
```
for &ethernet. Shouldn't it be just set to
`mtd-mac-address = <&iNIC_rf 0x4>;`
as it gets the same address as 2.4 GHz Wifi?

Regarding lan_mac, you can most probably just keep it. I have an unresolved discussion about that elsewhere (why do we set the bridge address here when only eth0 is in bridge and that one has the same MAC address), but this has to be answered in general before you have to care about it.

adrianschmutzler commented:
Ah, and can you read some additional bytes from the factory partition? Sometimes a second MAC address is stored at 40 or 46.

nbriggs commented:
Re the ethernet address -- I did that because I wasn't happy with it using the same HWaddr on multiple interfaces.  But it's not actually a problem, so I'll fix it as you suggested to pull from the iNIC_rf memory.

Re the additional bytes -- no other recognizable MAC addresses in there:

```
$ head mtd2-factory.hex
00000000  92 55 07 01 88 dc 96 03  c2 55 92 55 14 18 01 80  |.U.......U.U....|
00000010  00 00 92 55 14 18 00 00  01 00 6a ff 13 02 ff ff  |...U......j.....|
00000020  ff ff c1 ff 01 92 ff ff  ff ff ff ff ff ff ff ff  |................|
00000030  8e 75 01 43 22 ff 25 00  ff ff 14 ff ff ff d9 aa  |.u.C".%.........|
00000040  cc 88 ff ff 08 07 00 00  00 07 00 00 00 07 ff ff  |................|
00000050  ff 00 05 05 05 05 05 05  05 05 05 05 05 05 05 05  |................|
00000060  00 00 00 00 00 00 00 00  00 00 00 00 00 00 42 73  |..............Bs|
00000070  f7 00 86 0f 00 00 72 fc  10 10 10 10 10 10 10 10  |......r.........|
00000080  10 10 10 10 0f 0f 0f 0f  0f 0f 0f 10 10 11 11 12  |................|
00000090  12 13 14 10 10 10 11 12  12 13 14 ff ff ff ff ff  |................|
```
and
```
$ head mtd3-iNIC_rf.hex
00000000  20 76 04 01 88 dc 96 03  c2 54 ff ff ff ff ff ff  | v.......T......|
00000010  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
00000020  ff ff ff ff ff ff ff ff  00 0c 43 76 20 77 00 0c  |..........Cv w..|
00000030  43 76 20 66 22 0c 20 00  ff ff 49 01 55 77 a8 aa  |Cv f". ...I.Uw..|
00000040  8c 88 ff ff 0a 00 00 00  00 00 00 00 00 00 ff ff  |................|
00000050  ff ff 12 12 12 12 12 12  12 12 12 12 12 12 13 13  |................|
00000060  15 15 15 15 15 15 15 15  15 15 15 15 16 16 80 ff  |................|
00000070  ff ff 80 ff ff ff 00 00  ff ff ff ff ff ff ff ff  |................|
00000080  ff ff ff ff ff ff ff ff  ff ff ff ff ff ff ff ff  |................|
*
```

adrianschmutzler commented:
> Re the additional bytes -- no other recognizable MAC addresses in there
Thanks.

> I did that because I wasn't happy with it using the same HWaddr on multiple interfaces.

Never do this! Only use the addresses found in the vendor setup.
I've actually just recently observed two devices with consecutive MAC addresses (device1 = device2 + 2) in the same layer-2 network....

If you do the update as said the topic should be finished then.

nbriggs committed:
ramips: add support for EnGenius ESR600

The EnGenius ESR600 is a dual band wireless router with a 4-port gigabit
Ethernet switch, a gigabit Ethernet WAN port and a USB port.

Specification:

- Bootloader:	U-Boot
- SoC:		MediaTek MT7620A (600 MHz)
- Flash:	16MB, Macronix MX25L12845E
- RAM:		64MB, Nanya NT5TU32M16DG-AC
- Serial:	115200 baud, no header, 3.3V
  		J2: Vcc (arrow), Gnd, Tx, Rx
- USB:		USB 2, 5V
- Ethernet:	5 x 1 Gb/s 4 LAN 1 WAN, Atheros AR8327
- WiFi0:	5 GHz 802.11 b/g/n Ralink RT5592N
  		300 Mb/s, 2T2R
- WiFi1:	2.4 GHz 802.11 b/g/n integrated
  		300 Mb/s, 2T2R
- Antennas:	2 per radio, internal
- LEDs:		1 programmable power (amber)
  		2 programable radio (blue)
		1 programable WPS-5G (blue)
		1 non-programable WAN activity (blue)
		1 unconfigured WPS-2.4G (amber)
- Buttons:	GPIO: Reset, WPS

Installation:

Use the OEM web interface to install the ...-factory.dlf image.
Use the OpenWRT ...-sysupgrade.bin image for future upgrades.

The J2 serial port can be accessed either by soldering in a header,
standard 0.1" spacing, or by using pogo-pins against the back side.

As configured by the OEM, the U-Boot boot delay is short, however quickly
typing "1" leads to the U-Boot "System load Linux to SDRAM via TFTP"
prompt.  The TFTP client is configured by default with
     client:   192.168.99.9
     server:   192.168.99.8
     filename: uImageESR600
It will load an OpenWRT initramfs kernel with this method.

Known issues:

1) Only the ports externally labeled WAN, LAN3 and LAN4 are operational.
LAN1 and LAN2 do not appear to power up. This issue is also present
in the Lava LR25G001.

2) The amber WPS-2.4G LED, in the same lightguide as the blue WPS-5G LED,
is not configured in the Device Tree specification.

3) The blue WAN activity LED is not configured in the Device Tree
specification as this causes the AR8327 switch to fail to initialize.

Signed-off-by: Nick Briggs <nicholas.h.briggs@gmail.com>

nbriggs commented:
Resolved issue with ethernet MAC address, now matches 2.4GHz radio address per @adrianschmutzler suggestion.

adrianschmutzler commented on target/linux/ramips/base-files/etc/board.d/01_leds:
```diff
@@ -187,6 +187,10 @@ elecom,wrh-300cr)
 	ucidef_set_led_netdev "lan" "lan" "$boardname:green:ethernet" "eth0"
 	;;
+engenius,esr600)
+	ucidef_set_led_netdev "wlan5g" "5.0GHz" "$boardname:blue:wlan5g" "wlan0"
```
I'd personally use just "5GHz" there instead of "5.0GHz". I think the latter is used rarely. But it's a matter of taste.

nbriggs replied:
I would have done that too, but the case label for that LED is "5.0GHz", so I went with that.

adrianschmutzler replied:
Okay, fine.

nbriggs commented:
@adrianschmutzler ... question: what's the right way to resolve the conflicting files noted above that will leave my branch in a state that can be merged, complying with the 1-commit rule?  If I resolve it within github, I think I'll end up with multiple commits -- but should I do that, then pull into my local version, squash the extra commits and then force push the branch back?  Thanks for any advice you can give on this!

adrianschmutzler commented:
I do not know how to resolve it "in GitHub", but do it like this:
You have some local branch containing the PR changes somewhere, from where you pushed to GitHub when creating the PR. Let's call it "localbranch".
In your local repository, first fetch the latest changes from the official repo (fetch, not pull):
`git fetch upstream master`
This assumes that you have linked the official repo as remote, e.g. with
`git remote add upstream https://github.com/openwrt/openwrt/` earlier.
Now check that you are in your `localbranch`, otherwise do a `git checkout localbranch`. Now "rebase" your local branch on top of openwrt master:
`git rebase upstream/master`
This will first go to current state of "upstream/master" (you just fetched) and then apply each of the patches in `localbranch` on top. The result will be your `localbranch` again, still checked out.
You might have to fix rebase conflicts in between.
Finally, push your updated `localbranch` to your Github repository (assuming that's `origin`), overwriting previous changes:
`git push -f origin localbranch`
This will update your PR automatically, assuming that localbranch is the correct name.

nbriggs commented:
@adrianschmutzler -- perfect, thanks.

nbriggs commented:
@ynezz Hi Petr -- I think that all the requested changes have been resolved, is there anything else that you believe needs to be done before this PR can be merged?

adrianschmutzler commented on target/linux/ramips/dts/mt7620a_engenius_esr600.dts:
```diff
@@ -0,0 +1,211 @@
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x4000000>;
+	};
```
I think memory node can be dropped.

nbriggs replied:
I'm away from the device until after 17 August, but I will make that change and  test as soon as possible after that.

nbriggs replied:
I can confirm that it auto-detects the correct memory size without the "memory@0" node in the DTS.  I will remove that.

adrianschmutzler commented on target/linux/ramips/base-files/etc/board.d/02_network:
```diff
@@ -329,6 +329,10 @@ ramips_setup_interfaces()
 		;;
+	engenius,esr600)
+	    ucidef_add_switch "switch0" \
+			"1:lan:4" "2:lan:3" "3:lan:2" "4:lan:1" "5:wan" "0t@eth0"
```
Does it really need to be "0t@eth0" here instead of "0@eth0"? I thought both would give the same result?

nbriggs replied:
The answer that it appears to work equally well with either "0t@eth0" or "0@eth0".
Since "0@eth0" appears to be the preferred notation I'll switch to that.

adrianschmutzler commented on target/linux/ramips/dts/mt7620a_engenius_esr600.dts:
```diff
@@ -0,0 +1,211 @@
+		compatible = "pci1814,5592";
+		reg = <0x0 0 0 0 0>;
+		ralink,mtd-eeprom = <&factory 0x0>;
+		mtd-mac-address = <&factory 0x4>;
```
If you use `ralink,mtd-eeprom = <&factory 0x0>;`, the MAC address should automatically read from 0x0 + 4, so <&factory 0x4> . Please try if you still get a correct MAC address if you just remove mtd-mac-address here.

nbriggs replied:
Yes, it gets the correct MAC address when the mtd-mac-address clause is removed.

adrianschmutzler commented on target/linux/ramips/dts/mt7620a_engenius_esr600.dts:
```diff
@@ -0,0 +1,211 @@
+
+&wmac {
+	ralink,mtd-eeprom = <&iNIC_rf 0x0>;
+	mtd-mac-address = <&iNIC_rf 0x4>;
```
Same as above, try whether you still get the correct address if you remove the mtd-mac-address line.

nbriggs replied:
Yes, it gets the correct MAC address when the mtd-mac-address clause is removed.

adrianschmutzler commented on target/linux/ramips/base-files/etc/board.d/02_network:
```diff
@@ -329,6 +329,10 @@ ramips_setup_interfaces()
 			"1:lan" "2:lan" "3:lan" "4:lan" "0:wan" "6@eth0"
 		;;
+	engenius,esr600)
+	    ucidef_add_switch "switch0" \
```
Wrong indent here (spaces) ...

nbriggs replied:
Fixed, pushed.

adrianschmutzler commented:
@ynezz @blocktrron Despite the indent to be fixed, this should be ready for another review now.

nbriggs commented on target/linux/ramips/dts/mt7620a_engenius_esr600.dts:
```diff
@@ -0,0 +1,204 @@
+		reg = <0>;
+		spi-max-frequency = <50000000>;
+		m25p,fast-read;
+	
```
Whitespace error: trailing tab on this line.

nbriggs replied:
Fixed whitespace error.  No other changes.

adrianschmutzler commented:
@nbriggs That's beyond my knowledge, can't comment on it.

nbriggs commented:
Rebased/pushed to resolve conflict for target/linux/ramips/base-files/etc/board.d/02_network

ynezz commented:
Thanks merged into my staging tree.
